### PR TITLE
Enhanced RDoc for CSV::Table

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Methods:
- delete
- delete_if
- each
- ==

### CODE REVIEW

File: lib/csv/table.rb [hunk 1]
```diff
@@ -792,11 +792,48 @@ def push(*rows)
...
+    #
+    # In either case, the returned values are in the order
+    # specified by the arguments.  Arguments may be repeated.
     #
-    # Removes and returns the indicated columns or rows. In the default mixed
-    # mode indices refer to rows and everything else is assumed to be a column
-    # headers. Use by_col!() or by_row!() to force the lookup.
+    # ---
+    #
+    # Returns rows as an \Array of \CSV::Row objects:
```
Comment (esparta) [inline lib/csv/table.rb:809]: I'm a little confused here; does `delete` need the example/explanation of `by_row!` ?

File: lib/csv/table.rb [hunk 2]
```diff
@@ -792,11 +792,48 @@ def push(*rows)
...
+    #   values # => [#<CSV::Row "Name":"foo" "Value":"0">]
+    #
+    # Two indexes:
+    #   table = CSV.parse(source, headers: true)
+    #   values = table.delete(2, 0)
+    #   values # => [#<CSV::Row "Name":"baz" "Value":"2">, #<CSV::Row "Name":"foo" "Value":"0">]
+    #
+    # ---
+    #
+    # Returns columns data as column Arrays:
```
Comment (esparta) [inline lib/csv/table.rb:825]: Same for `by_row!`, maybe a reference would be enough if needed?

File: lib/csv/table.rb [hunk 3]
```diff
@@ -821,16 +858,32 @@ def delete(*indexes_or_headers)
       end
     end
 
+    # Removes rows or columns for which the block retruns a truthy value;
```
Comment (esparta) [inline lib/csv/table.rb:861]: small typo:

```suggestion
    # Removes rows or columns for which the block returns a truthy value;
```

File: lib/csv/table.rb [hunk 4]
```diff
@@ -792,11 +792,48 @@ def push(*rows)
...
+    #
+    # One index:
+    #   source = "Name,Value\nfoo,0\nbar,1\nbaz,2\n"
+    #   table = CSV.parse(source, headers: true)
+    #   values = table.delete(0)
+    #   values # => [#<CSV::Row "Name":"foo" "Value":"0">]
+    #
+    # Two indexes:
+    #   table = CSV.parse(source, headers: true)
+    #   values = table.delete(2, 0)
```
Comment (esparta) [inline lib/csv/table.rb:820]:
```suggestion
    #   deleted_rows= table.delete(2, 0)
```

File: lib/csv/table.rb (hunk 1 above)
Comment (BurdetteLamar) [reply lib/csv/table.rb]: The <tt>by_row!</tt> call was not needed in the first place.  Removed.

File: lib/csv/table.rb (hunk 2 above)
Comment (BurdetteLamar) [reply lib/csv/table.rb]: Same as above.  Removed.

File: lib/csv/table.rb (hunk 3 above)
Comment (BurdetteLamar) [reply lib/csv/table.rb]: Fixed.

File: lib/csv/table.rb (hunk 4 above)
Comment (BurdetteLamar) [reply lib/csv/table.rb]: Fixed (in several places).

**Comment by BurdetteLamar** [comment]
@esparta, the re-request thingy did not work.  Please re-review.

**Comment by BurdetteLamar** [comment]
@kou, I think this is ready for you.

**Comment by esparta** [comment]
> the re-request thingy did not work. Please re-review.

Hi @BurdetteLamar, I just re-review and I have no more comments.
Thanks!

**Comment by kou** [comment]
Thanks!

**Pull request merged**